soc/arm/st_stm32/stm32l5/Kconfig.defconfig.stm32l562xx

```
@@ -0,0 +1,14 @@
+# ST Microelectronics STM32L562XX MCU
+
+# Copyright (c) 2020 Yestin Sun
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32L562XX
+
+config SOC
+	default "stm32l562xx"
+
+config NUM_IRQS
+	default 108
```

```suggestion
	default 109
```
This value is used as array size

---

OK, I read the datasheets for both STM32L552xx and STM32L562xx. The maximum maskable interrupt channels are both 109. So wonder why it is 108 in the stm32l552xx defconfig?

---

According to https://raw.githubusercontent.com/zephyrproject-rtos/hal_stm32/master/stm32cube/stm32l5xx/soc/stm32l552xx.h, maximum IRQ number is 107.
There might be a misalignment between ref man and Cube package then.
Even though this is highly theoretical, as max array size might be reached only if all IRQ are configured in same binary, from Zephyr perspective, we're CMSIS file as reference so value for STM32L552xx should be 107+1, until is is potentially fixed.

---

Ah, I see. Thanks for the explanation.